store/httpstore.go

```diff
@@ -99,10 +99,6 @@ func (s HTTPStore) GetMeta(name string, size int64) ([]byte, error) {
 	logrus.Debugf("%d when retrieving metadata for %s", resp.StatusCode, name)
 	b := io.LimitReader(resp.Body, size)
 	body, err := ioutil.ReadAll(b)
-	if resp.ContentLength > 0 && int64(len(body)) < resp.ContentLength {
```

Comment:
Oh, ok.  I don't know about the important of this check.  But if you wanted a check on sizes, maybe it would sense to check this in `downloadSigned`?  If there is an expected size (which could work like expectedSha256), it could verify that the size matches rather than rely on the content length header).